### PR TITLE
Add edge cases to beta binomial and tests

### LINKED ISSUE #114: Issue with degenerate beta binomial?
In the limit of b -> 0, we expect (AFAIK) the beta binomial distribution to approach a delta distribution with E(x) = n a / (a + b) ->  n.  However, I find that the log likelihood of the beta binomial actually gives a discrete uniform when b=0.  See driver code below and output figures of the pmf.  

```
import numpy as np
import pymc as pm

a = 10.
n = 4
xgrid = np.arange(5)

for i, b in enumerate([1E-2, 1E-5, 0.0]):
  logprobs = np.array([pm.betabin_like(x, a, b, n) for x in xgrid])
  probs = np.exp(logprobs)
  plt.figure()
  plt.title(str(b))
  plt.plot(xgrid, probs, label=str(b))
  plt.ylim(-0.1, 1.1)
  plt.savefig("./%d.png" % i)

```

image

image

image

### PR DESCRIPTION
**Pull request by kyleabeauchamp**
Should fix #114 

I'm not a fortran person though so YMMV.

### CODE REVIEW

File: pymc/flib.f [hunk 1]
```diff
@@ -4221,22 +4221,26 @@ SUBROUTINE betabin_like(x,alpha,beta,n,nx,na,nb,nn,like)
         if (na .NE. 1) atmp = alpha(i)
         if (nb .NE. 1) btmp = beta(i)
         if (nn .NE. 1) ntmp = n(i)
-        if ((atmp.LE.0.0).OR.(btmp.LE.0.0).OR.(ntmp.LE.0)) then
+        if (((atmp.LE.0.0).AND.(btmp.LE.0.0)).OR.(ntmp.LE.0)) then
           like = -infinity
           RETURN
-        endif
-        if (x(i) .LT. 0) then
+        else if (x(i) .LT. 0) then
```
Comment (kyleabeauchamp) [inline pymc/flib.f:4227]: I noticed that there's an explicit check here for x[i] < 0 but no corresponding check for x[i] > n.  E.g. consider the following:

```
In [8]: pm.betabin_like(7, 5, 0, 5)
Out[8]: 0.0

In [9]: pm.betabin_like(-1, 5, 0, 5)
Out[9]: -1.7976931348623157e+308
```

Comment (kyleabeauchamp) [inline pymc/flib.f:4227]: Added in latest commit.

**Comment by fonnesbeck** [comment]
Great, this looks fine. Thanks.

File: pymc/flib.f [hunk 2]
```diff
@@ -4221,22 +4221,29 @@ SUBROUTINE betabin_like(x,alpha,beta,n,nx,na,nb,nn,like)
...
           like = -infinity
           RETURN
-        endif
-        like =like + gammln(atmp+btmp)
-        like =like - gammln(atmp) - gammln(btmp)
+        else if (x(i) .GT. ntmp) then
+          like = -infinity
+          RETURN
+        else if ((atmp.LE.0.0).AND.(x(i).LE.0)) then
```
Comment (fonnesbeck) [inline pymc/flib.f:4233]: Actually, looking more closely at these, its not clear why having negative `a` and `b` values should return anything but a -inf. I can understand handling zero for degenerate cases, but the beta-binomial should never accept negative parameter values.
